Example/iOS-Example/Model+Fetchable.swift

```diff
@@ -20,7 +20,9 @@ extension FetchableObjectProtocol where Self: Model {
 
         return FetchDefinition<Self>(
             request: { completion in
-                completion(fetchAll())
+                DispatchQueue.main.async {
```

Comment:
@lickel This will make the swiftUI example work as expected.

Reply:
Alternatively I was able to address the issue modifying the FetchableRequest, by wrapping the fetch definition like so
``
```
 public init(
        definition: FetchDefinition<FetchedObject>,
        sortDescriptors: [NSSortDescriptor] = [],
        debounceInsertsAndReloads: Bool = true,
        animation: Animation? = nil
    ) {
        let controller = FetchedResultsController(
            definition: FetchDefinition<FetchedObject>(
                request:{ completion in
                    DispatchQueue.main.async {
                        definition.request(completion)
                    }
                },
                objectCreationToken: definition.objectCreationToken,
                creationInclusionCheck: definition.creationInclusionCheck,
                associations: definition.associations,
                dataResetTokens: definition.dataResetTokens)
            ,
            sortDescriptors: sortDescriptors,
            debounceInsertsAndReloads: debounceInsertsAndReloads
        )

        self.init(controller: controller, animation: animation)
    }
```

But I will leave that solution up to you.

Reply:
yup makes sense. I think I'd still like to fix it somewhere in the framework, but thanks!